src/cat.c

```diff
@@ -72,6 +72,9 @@ static void reset_state(struct cat_object *self)
         }
         self->cmd = NULL;
         self->cmd_type = CAT_CMD_TYPE_NONE;
+        #if CONFIG_LIB_CAT_USRDATA
```

Comment:
at this moment in cAT library there is no macro-based configuration feature. adding this kind of feature should be separated task.